### ISSUE TITLE
Unable to provision private AKS Cluster

### ISSUE THREAD

**Issue by annablender** [rancher/rancher#37395]
**Rancher Server Setup**
- Rancher version: 2.6.3
- Installation option (Docker install/Helm Chart): Docker install with `--debug`
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.21.9
- Cluster Type (Local/Downstream): 
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): 3 node downstream private AKS Cluster, provisioned through the Rancher UI

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) admin
  - If custom, define the set of permissions:
   
**Describe the bug**

I am unable to successfully provision a private AKS cluster using the Rancher UI on Rancher 2.6.3. I tried three times on a fresh docker install.

Try 1
![image](https://user-images.githubusercontent.com/12955547/164029468-ba1b9e03-4c52-49e2-a54d-506ba6248279.png)

Try 2 - appears cluster was in a failed state in Azure
![image](https://user-images.githubusercontent.com/12955547/164029523-d1806a5d-f881-4cae-bb6b-ed4999e6bba9.png)

Try 3 - appears now that Azure never created the nodes at all. There was no evidence of this cluster in Azure portal
![image](https://user-images.githubusercontent.com/12955547/164029638-5cf8768b-2b96-49d3-990f-8dd4bc23779d.png)

**To Reproduce**
 

* Run Rancher 2.6.3 in `--debug` mode on a docker install
* Login as admin
* Add Azure cloud credentials
* Create a 3 node, private downstream cluster using the Rancher UI. Save the import commands. (normally here, I would run the import command on a bastion VM in the same vnet as the downstream cluster to import the private AKS Cluster into Rancher. But, I was not able to reach that point)

**Result**

Unable to provision private AKS Cluster.

**Expected Result**

Private cluster provisions successfully in AKS and is ready for import.

**Comment by annablender** [comment]
Seeing the same thing as Try 3 with `Waiting for API to be available message` on rancher 2.6-head with kubernetes 1.22.6. Same repro steps.

Docker logs
```
2022/04/19 15:37:38 [DEBUG] could not get kubeconfig for cluster c-nbl6w
2022/04/19 15:37:38 [INFO] waiting for cluster AKS [c-nbl6w] to update
2022/04/19 15:37:38 [DEBUG] DesiredSet - No change(2) provisioning.cattle.io/v1, Kind=Cluster fleet-default/c-nbl6w for provisioning-cluster-create c-nbl6w
2022/04/19 15:37:43 [DEBUG] AKS cluster [c-nbl6w] will be managed by aks-operator-controller, skipping update
2022/04/19 15:37:43 [DEBUG] could not get kubeconfig for cluster c-nbl6w
2022/04/19 15:37:43 [INFO] waiting for cluster AKS [c-nbl6w] to update
2022/04/19 15:37:43 [DEBUG] DesiredSet - No change(2) provisioning.cattle.io/v1, Kind=Cluster fleet-default/c-nbl6w for provisioning-cluster-create c-nbl6w
2022/04/19 15:37:48 [DEBUG] AKS cluster [c-nbl6w] will be managed by aks-operator-controller, skipping update
2022/04/19 15:37:48 [DEBUG] could not get kubeconfig for cluster c-nbl6w
2022/04/19 15:37:48 [INFO] waiting for cluster AKS [c-nbl6w] to update
2022/04/19 15:37:48 [DEBUG] DesiredSet - No change(2) provisioning.cattle.io/v1, Kind=Cluster fleet-default/c-nbl6w for provisioning-cluster-create c-nbl6w
2022/04/19 15:37:53 [DEBUG] AKS cluster [c-nbl6w] will be managed by aks-operator-controller, skipping update
2022/04/19 15:37:53 [DEBUG] could not get kubeconfig for cluster c-nbl6w
2022/04/19 15:37:53 [INFO] waiting for cluster AKS [c-nbl6w] to update
2022/04/19 15:37:53 [DEBUG] DesiredSet - No change(2) provisioning.cattle.io/v1, Kind=Cluster fleet-default/c-nbl6w for provisioning-cluster-create c-nbl6w
2022/04/19 15:37:58 [DEBUG] AKS cluster [c-nbl6w] will be managed by aks-operator-controller, skipping update
2022/04/19 15:37:58 [DEBUG] could not get kubeconfig for cluster c-nbl6w
2022/04/19 15:37:58 [INFO] waiting for cluster AKS [c-nbl6w] to update
2022/04/19 15:37:58 [DEBUG] DesiredSet - No change(2) provisioning.cattle.io/v1, Kind=Cluster fleet-default/c-nbl6w for provisioning-cluster-create c-nbl6w
```

**Comment by thedadams** [comment]
In order to consolidate the issues we are working on, the issues referenced here will be fixed as a part of https://github.com/rancher/rancher/issues/37164. Closing as duplicate.
